versions-common/src/main/java/org/codehaus/mojo/versions/api/PomHelper.java

```diff
@@ -1362,48 +1334,39 @@ public static ProjectBuildingRequest createProjectBuilderRequest(
-        result.put(path, model);
-        result.putAll(getReactorModels(path, model, project, logger));
+        result.put(project.getFile(), model);
+        result.putAll(getChildModels(model, project, logger));
         return result;
     }
 
     /**
      * Builds a sub-map of raw models keyed by module path.
      *
-     * @param path    The relative path to base the sub-map on.
-     * @param model   The model at the relative path.
+     * @param model   The root model
      * @param project The project to build from.
      * @param logger  The logger for logging.
      * @return A map of raw models keyed by path relative to the project's basedir.
      * @throws IOException if things go wrong.
      */
-    private static Map<String, Model> getReactorModels(String path, Model model, MavenProject project, Log logger)
-            throws IOException {
-        Map<String, Model> result = new LinkedHashMap<>();
-        Map<String, Model> childResults = new LinkedHashMap<>();
-        Set<String> childModules = getAllChildModules(model, logger);
-
-        File baseDir = path.length() > 0 ? new File(project.getBasedir(), path) : project.getBasedir();
-        removeMissingChildModules(logger, baseDir, childModules);
+    private static Map<File, Model> getChildModels(Model model, MavenProject project, Log logger) throws IOException {
+        Map<File, Model> result = new LinkedHashMap<>();
+        Map<File, Model> childResults = new LinkedHashMap<>();
 
-        childModules.stream()
+        File baseDir = model.getPomFile().getParentFile();
+        getAllChildModules(model, logger).parallelStream()
                 .map(moduleName -> new File(baseDir, moduleName))
+                .map(file -> file.isFile() ? file : new File(file, "pom.xml"))
                 .filter(File::exists)
-                .forEach(moduleFile -> {
-                    File pomFile = moduleFile.isDirectory() ? new File(moduleFile, "/pom.xml") : moduleFile;
-                    String modulePath = (!path.isEmpty() && !path.endsWith("/") ? path + "/" : path)
-                            + pomFile.getParentFile().getName();
-
+                .forEach(pomFile -> {
                     try {
                         // the aim of this goal is to fix problems when the project cannot be parsed by Maven,
                         // so we have to work with the raw model and not the interpolated parsed model from maven
                         Model moduleModel = getRawModel(pomFile);
-                        result.put(modulePath, moduleModel);
-                        childResults.putAll(getReactorModels(modulePath, moduleModel, project, logger));
+                        result.put(pomFile, moduleModel);
+                        childResults.putAll(getChildModels(moduleModel, project, logger));
                     } catch (IOException e) {
                         logger.debug("Could not parse " + pomFile.getPath(), e);
                     }
```

Comment:
At least should be an error, maybe we should not cache it.
Now with a problem - build status is ok - but not every module are updated.

Reply:
Well, I did not want to change the existing behaviour. Should we then introduce a toggle for that? Or straight on throw an error?

Reply:
@slawekjaranowski applied the suggested change, please advise on the above